biz.aQute.bndlib.tests/test/test/jpms/JPMSModuleInfoPluginTest.java

```diff
@@ -26,26 +27,43 @@
 
 public class JPMSModuleInfoPluginTest {
 
+	@ParameterizedTest(name = "Validate Module Access inputs with invalid content {arguments}")
+	@CsvSource(delimiterString = "|", value = {
```

Comment:
The delimiter value is not present in the source values. If your intention is not to split the values, then use the `ValueSource` annotation instead of `CsvSource`.

Reply:
got it

Reply:
Actually, bit of a tip: just use CsvSource without specifying the delimiter. Then you can also drop the "value=" and make it more compact. ValueSource forces you to specify strings= and so is less compact. I always use CsvSource for strings. Unless BJ knows of a good reason to avoid this practice that I'm not aware of...

Reply:
You can use CsvSource for simple strings, but its purpose is to split a string into multiple values at the delimiter. If you don't intend that, then using CsvSource does not properly communicate your intent to future developers looking at your code. While using ValueSource requires you to use the strings element, it does properly communicate your intent.

I made this comment because I was confused. So the intent was not clear in using CsvSource here.

Reply:
The number and type of the arguments to the test method are a bit of a giveaway as to what values you're expecting. I think the more confusing thing was that a delimiter was explicitly specified when none was intended. But I'm not super-fussed and I guess requiring us to add "strings=" is not the end of the world. :smile:

Reply:
The delimiter had to be specified since the string values include commas which are the default delimiter.

Reply:
👍 Of course.